tests/Factories/AdminControllerFactoryTest.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use Portal\Factories\AdminControllerFactory;
+use Slim\Views\PhpRenderer;
+
+class AdminControllerFactoryTest extends TestCase
+{
+    function testInvoke()
+    {
+        $stub = $this->createMock(PhpRenderer::class);
```

Comment:
similar to the tests/Factories/LoginControllerFactoryTest.php, you should be replicating the real factory:
$container = $this->createMock(ContainerInterface::class);
then
$container->method('get')->willReturn...

Reply:
done